google/cloud/storage/client.py

```diff
@@ -897,7 +907,7 @@ def create_bucket(
             project = self.project
 
         # Use no project if STORAGE_EMULATOR_HOST is set
-        if _BASE_STORAGE_URI not in _get_storage_host():
+        if _get_storage_host() != _DEFAULT_STORAGE_HOST:
```

Comment:
Is this a change that will impact backwards compatibility?

Reply:
This is a special case only suited for when `STORAGE_EMULATOR_HOST` is set. This fix ensures the methods operate the same for custom endpoints and storage default endpoint.